scaleway/helpers.go

```diff
@@ -281,7 +281,25 @@ func getZone(d terraformResourceData, meta *Meta) (utils.Zone, error) {
 		return zone, nil
 	}
 
-	return utils.Zone(""), fmt.Errorf("could not detect region")
+	return utils.Zone(""), fmt.Errorf("could not detect zone")
+}
+
+// getRegion will try to guess the region from the following:
+//  - region field of the resource data
+//  - default region from config
+func getRegion(d terraformResourceData, meta *Meta) (utils.Region, error) {
+
+	rawRegion, exist := d.GetOkExists("region")
+	if exist {
+		return utils.ParseRegion(rawRegion.(string))
+	}
+
+	region, exist := meta.scwClient.GetDefaultRegion()
+	if exist {
+		return region, nil
+	}
+
+	return utils.Region(""), fmt.Errorf("could not detect region")
```

Comment:
Nit: since the error does not have dynamic data consider declaring it:

ErrRegionNotFound 

Or similar